pkg/pool-stable-phantom/contracts/StablePoolProtocolFees.sol

```diff
@@ -156,6 +150,52 @@ abstract contract StablePoolProtocolFees is StablePoolStorage, StablePoolRates,
             );
     }
 
+    function _getGrowthInvariants(uint256[] memory balances, uint256 lastPostJoinExitAmp)
+        internal
+        view
+        returns (
+            uint256 swapFeeGrowthInvariant,
+            uint256 totalNonExemptGrowthInvariant,
+            uint256 totalGrowthInvariant
+        )
+    {
+        // We always calculate the swap fee growth invariant, since we cannot easily know whether swap fees have
+        // accumulated or not.
+
+        swapFeeGrowthInvariant = StableMath._calculateInvariant(
+            lastPostJoinExitAmp,
+            _getAdjustedBalances(balances, true) // Adjust all balances
+        );
+
+        // For the other invariants, we can potentially skip some work. In the edge cases where none or all of the
+        // tokens are exempt from yield, there's one fewer invariant to compute.
+
+        if (_areNoTokensExempt()) {
+            // If there are no tokens exempt of yield fee, then the total non exempt growth will equal the total growth
+            // (because all yield growth is non exempt). There's also no point in adjusting balances to get the
+            // non-exempt ones, as none are exempt and as such this equals the current balances.
```

Comment:
```suggestion
            // If there are no tokens with fee-exempt yield, then the total non-exempt growth will equal the total
            // growth: all yield growth is non-exempt. There's also no point in adjusting balances, since we
            // already know none are exempt.
```